File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        left = [i for i in nums]
        right = [j for j in nums]
        size = len(nums)

        for x in range(size):

            if x > 0 : 
                left[x] *= left[x - 1]
                
                right[size - x - 1] *= right[ size - x ]

                

        print(right)
        print(left)
        for l in range(size):

            if l == 0 :
                nums[l] = right[l + 1]
            elif l == size - 1:
                nums[l] = left[l - 1]
            else :
                nums[l] = left[l - 1] * right[l + 1]

        return nums                                

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0 ; 
        int right = heights.length - 1 ;
        int maxSize = 0 ; 
        int currSize = 0 ;
        while (left < right ) {
            int width = right - left ;
            int height = Math.min(heights[left] , heights[right]);
            currSize = width * height ;

            maxSize = Math.max(currSize , maxSize);
            if (height == heights[right])right--;
            else{
                left ++ ;
            }
        }
        return maxSize ;

    }
}
